test/functional/launch/nested_direct/tests/test-launch-nested-Direct.hpp

```diff
@@ -0,0 +1,192 @@
+  constexpr int threads_y = 3;
+  constexpr int threads_z = 4;
+
+  constexpr int blocks_x = 4;
+  constexpr int blocks_y = 5;
+  constexpr int blocks_z = 6;
+
+  if ( RAJA::stripIndexType(N) > 0 ) {
+
+    std::iota(test_array, test_array + RAJA::stripIndexType(N), 0);
+
+    constexpr int DIM = 6;
+    using layout_t = RAJA::Layout<DIM, INDEX_TYPE,DIM-1>;
+    RAJA::View<INDEX_TYPE, layout_t> Aview(working_array, N6, N5, N4, N3, N2, N1);
+
+    RAJA::launch<LAUNCH_POLICY>
+      (RAJA::LaunchParams(RAJA::Teams(blocks_x, blocks_y, blocks_z), RAJA::Threads(threads_x, threads_y,threads_z)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+        RAJA::loop<TEAM_Z_POLICY>(ctx, r6, [&](INDEX_TYPE bz) {
+            RAJA::loop<TEAM_Y_POLICY>(ctx, r5, [&](INDEX_TYPE by) {
+                RAJA::loop<TEAM_X_POLICY>(ctx, r4, [&](INDEX_TYPE bx) {
+
+                    RAJA::loop<THREAD_Z_POLICY>(ctx, r3, [&](INDEX_TYPE tz) {
+                        RAJA::loop<THREAD_Y_POLICY>(ctx, r2, [&](INDEX_TYPE ty) {
+                            RAJA::loop<THREAD_X_POLICY>(ctx, r1, [&](INDEX_TYPE tx) {
+
+                                auto idx = tx + N1 * (ty + N2 * (tz + N3 * (bx + N4 * (by + N5 * bz))));
+
+
+                                Aview(bz, by, bx, tz, ty, tx) = static_cast<INDEX_TYPE>(idx);
+                              });
+                          });
+                      });
+
+                  });
+              });
+          });
+    });
+  } else { // zero-length segment
+
+    memset(static_cast<void*>(test_array), 0, sizeof(INDEX_TYPE) * data_len);
+
+    working_res.memcpy(working_array, test_array, sizeof(INDEX_TYPE) * data_len);
+
+    RAJA::launch<LAUNCH_POLICY>
+      (RAJA::LaunchParams(RAJA::Teams(blocks_x, blocks_y, blocks_z), RAJA::Threads(blocks_x, blocks_y ,blocks_z)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+        RAJA::loop<TEAM_Z_POLICY>(ctx, r3, [&](INDEX_TYPE bz) {
+            RAJA::loop<TEAM_Y_POLICY>(ctx, r2, [&](INDEX_TYPE by) {
+                RAJA::loop<TEAM_X_POLICY>(ctx, r1, [&](INDEX_TYPE bx) {
+
+                    RAJA::loop<THREAD_Z_POLICY>(ctx, r3, [&](INDEX_TYPE tz) {
+                        RAJA::loop<THREAD_Y_POLICY>(ctx, r2, [&](INDEX_TYPE ty) {
+                            RAJA::loop<THREAD_X_POLICY>(ctx, r1, [&](INDEX_TYPE tx) {
+
+                                (void) tx;
+                                (void) ty;
+                                (void) tz;
+
+                                (void) bx;
+                                (void) by;
+                                (void) bz;
+
+                                working_array[0]++;
+                              });
+                          });
+                      });
+
+                  });
+              });
+          });
+      });
+  }
+
+  working_res.memcpy(check_array, working_array, sizeof(INDEX_TYPE) * data_len);
+
+  for (INDEX_TYPE i = INDEX_TYPE(0); i < N; i++) {
```

Comment:
if N is 0 does this test anything, should it at least test that nothing happened?

Reply:
That is a good point, I based it off what we did here: 

https://github.com/LLNL/RAJA/blob/develop/test/functional/forall/segment/tests/test-forall-RangeSegment.hpp#L61-L63

How can we test that nothing happened?

Reply:
I see that there is `working_array[0]++;` above, so you could test that  `working_array[0]` doesn't change. Assuming that its allocated and initialized when N is 0.

Reply:
good idea, I just pushed up the updates with the checks.